src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -27,8 +30,14 @@ object ResourceMatcher {
 
     def scalarMatch(name: String): Option[ScalarMatch] = scalarMatches.find(_.resourceName == name)
 
-    def resources: Iterable[org.apache.mesos.Protos.Resource] =
-      scalarMatches.flatMap(_.consumedResources) ++ portsMatch.resources
+    def resources: Iterable[Protos.Resource] =
+      scalarMatches.flatMap(_.consumedResources) ++
+        portsMatch.resources
+
+    // TODO - this assumes that volume matches are one resource to one volume, which should be correct, but may not be.
+    // TODO - don't use this because it doesn't return reservation info !!!
```

Comment:
I actually think I fixed this but didn't get rid of the TODO.  I was thinking that it might be possible to get multiple offers for the same disk resource (as in, half of the space already allocated to the principal, but the other was not allocated at all)